### PR TITLE
Milestone 11 : Api endpoints

### PR DESCRIPTION
**Pull request by aboongm**
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  #  def index
+  #   @user = current_user
+  #   @posts = @user.posts.includes(:comments)
+
+  #   respond_to do |format|
+  #     format.json { render json: @posts }
+  #   end
+  # end
```
Comment (Mihndim2020) [inline app/controllers/api/v1/posts_controller.rb]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  #  def index
+  #   @user = current_user
+  #   @comments = @user.comments
+
+  #   respond_to do |format|
+  #     format.json { render json: @comments }
+  #   end
+  # end
```
Comment (Mihndim2020) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] Please kindly remove `commented code` from your `code base`, as it makes your `code base` dirty and less readable. To make sure that your codebase is always clean, always remember to remove commented-codes.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi Team, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark: Correct use of Gitflow. 
:heavy_check_mark: No `Linter` errors.
:heavy_check_mark: Clean/Organized code.

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION


# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kingstalux — APPROVED** [review]
Hi @team,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  

Good job implementing the required changes :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights :clap: 
:ballot_box_with_check: Good commit history :+1: 
:ballot_box_with_check: Linters are passing :ok_hand: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**